src/main/java/seedu/address/logic/parser/AddAppointmentParser.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+
+import java.util.Calendar;
+import java.util.List;
+import java.util.stream.Stream;
+
+import com.joestelmach.natty.DateGroup;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.AddAppointmentCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parse input arguments and creates a new AddAppointmentCommand Object
+ */
+public class AddAppointmentParser implements Parser<AddAppointmentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddAppointmentCommand
+     * and returns an AddAppointmentCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public AddAppointmentCommand parse(String userInput) throws ParseException {
+
+        if (userInput.split(" ").length == 1) {
+            return new AddAppointmentCommand();
+        }
+
+        ArgumentMultimap argumentMultimap =
+                ArgumentTokenizer.tokenize(userInput, PREFIX_DATE);
+
+        if (!arePrefixesPresent(argumentMultimap, PREFIX_DATE)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddAppointmentCommand.MESSAGE_USAGE));
+        }
+
+        String[] args = userInput.split(" ");
+        try {
+
+            Index index = Index.fromOneBased(Integer.parseInt(args[1]));
+            com.joestelmach.natty.Parser parser = new com.joestelmach.natty.Parser();
+            List<DateGroup> groups = parser.parse(argumentMultimap.getValue(PREFIX_DATE).get());
```

Comment:
The _natty_ library is great, but make sure it parses dates in the correct format.